Source/WebCore/editing/mac/FrameSelectionMac.mm

```diff
@@ -40,9 +40,19 @@
     if (!AXObjectCache::accessibilityEnabled())
         return;
 
-    if (m_selection.start().isNotNull() && m_selection.end().isNotNull()) {
-        if (AXObjectCache* cache = m_document->existingAXObjectCache())
+    if (CheckedPtr cache = m_document->existingAXObjectCache()) {
+        if (m_selection.start().isNotNull() && m_selection.end().isNotNull())
             cache->postTextStateChangeNotification(m_selection.start(), intent, m_selection);
+        else {
+            // The selection was cleared, so use `onSelectedTextChanged` with an empty selection range to update the isolated tree.
+            // In the future, we should consider actually doing `postTextStateChangeNotification`, since right
+            // now, VoiceOver does not announce when text becomes unselected (because we don't post a notification),
+            // unless new text becomes selected at the same time. However, even if we did post this notification,
+            // changes would be needed in VoiceOver too, so just do onSelectedTextChanged for now.
+#if ENABLE(ACCESSIBILITY_ISOLATED_TREE)
+            cache->onSelectedTextChanged(m_selection);
+#endif
```

Comment:
VO doesn't need to announce that text is unselected, although it may do it in some cases. . But regardless of VO response in this case, we should fire the right notification.
Does this need to be iso tree specific? Shouldn't it work the same for all ports?

Reply:
> VO doesn't need to announce that text is unselected, although it may do it in some cases.

It announces "foo selected, bar unselected" when the selection changes, so I think reading "bar unselected" when bar is unselected without anything becoming selected at the same time could make sense and be useful for folks. Right now, the selection disappears without any announcement, which could maybe cause confusion for users.

> But regardless of VO response in this case, we should fire the right notification.
Agreed in general, but posting the right notification is a bit complicated:

1. VoiceOver expects the notification to be associated with an actual text object, and we don't have that information when text becomes unselected — the associated DOM node with an unselection is null. I tried posting from the root web area, but that does not work. So we would have to figure out how to pipe that information through, or start tracking the last selected object maybe?
2. It's not clear what `AXTextStateChangeType` unselection maps to — none are a good fit.

I personally don't think right now is the right time to fully unravel this thread, so I was hoping to make a bit of incremental progress and fix the ITM cache invalidation bug. But I could file a separate bug and add it here in a comment? What do you think? Do you think we should get all the way to the bottom here, or make a bug for follow-up and take the intermediate bug fix for now?

> Does this need to be iso tree specific? Shouldn't it work the same for all ports?
I'd say yes in general, however the definition of `AXObjectCache::onSelectedTextChanged` is gated by `PLATFORM(MAC)`. I could mention this in the follow-up bug, or happy to address in some other way if you have an idea!

Reply:
Andres and I spoke, and resolution is to file a separate bug with some details about the problems here.